tasks/files-to-granules/index.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const get = require('lodash.get');
+const path = require('path');
+
+const { getGranuleId } = require('@cumulus/cmrjs');
+const { parseS3Uri } = require('@cumulus/common/aws');
+const cumulusMessageAdapter = require('@cumulus/cumulus-message-adapter-js');
+
+/**
+ * Helper to turn an s3URI into a fileobject
+ * @param {string} s3URI s3://mybucket/myprefix/myobject.
+ * @returns {Object} file object
+ */
+function fileObjectFromS3URI(s3URI) {
+  const uriParsed = parseS3Uri(s3URI);
+  return {
+    name: path.basename(s3URI),
+    bucket: uriParsed.Bucket,
+    filename: s3URI,
+    fileStagingDir: path.dirname(uriParsed.Key)
+  };
+}
+
+/**
+ * Takes the files from input and granules and merges them into an object where
+ * each file is associated with it's granuleId.
+ *
+ * @param {Array} inputFiles - list of s3 files to add to the inputgranules
+ * @param {Array} inputGranules - an array of the granules
+ * @param {string} regex - regex needed to extract granuleId from filenames
+ * @returns {Object} an object that contains lists of each granules' files
+ *                   attatched by their granuleId
+ */
+function mergeInputFilesWithInputGranules(inputFiles, inputGranules, regex) {
```

Comment:
Feel free to use or ignore any part of this suggestion.

```js
const flatten = require('lodash.flatten');
const keyBy = require('lodash.keyby');

function mergeInputFilesWithInputGranules(inputFiles, inputGranules, regex) {
  const granulesHash = keyBy(inputGranules, (g) => g.granuleId);
  const filesFromInputGranules = flatten(inputGranules.map((g) => g.files));

  return inputFiles
    .filter((f) => !filesFromInputGranules.include(f))
    .forEach((f) => {
      granulesHash[getGranuleId(f, regex)].files.push(fileObjectFromS3URI(f));
    });
}
```

Reply:
👍 43321585116